demystifying_large_language_models/demystifying_large_language_models.md

```diff
@@ -0,0 +1,669 @@
+As promised, here's the TL;DR of this page:
+
+* **Neural networks** are systems that use lots of small calculations using **weights** that interact with one another to result in a final prediction.  A neural network (also known as a **deep learning** model) consists of an input layer which transmits input values to a hidden layer (or more than one) which analyzes inputs, and an output layer which collates results.
+* **Transformers** revolutionized language neural networks because they allowed more efficient access to context clues from further back in text than other systems allowed.  Today's LLMs rely on transformers.
+
+</div>
+
+Trial and Error
+---
+
+To understand neural networks, it's important to know that in machine learning (whether that's in a neural network or in another type of algorithm), there's a lot of trial and error.  In machine learning, the speed of calculation that a computer can offer allows the computer to try thousands of different values for the coefficient of a predictor variable, for example.  This allows the computer to determine which value ends up being the best coefficient, and do that for many different predictor (input) variables.
+
+![A figure with a hat states, "Oh, hey, you organized our photo archive!".  A second figure replies, "Yeah, I trained a neural net to sort the unlabeled photos into categories."  The first figure resonds, "Whoa! Nice work!."  The caption below the drawing reads: "Engineering tip: When you do a task by hand, you can technically say you trained a neural net to do it."](https://imgs.xkcd.com/comics/trained_a_neural_net.png "[Trained a Neural Net](https://xkcd.com/2173/) comic by xkcd, [CC BY-NC 2.5](https://xkcd.com/license.html).")<!-- style = "max-width: 220px;" -->Neural networks take this power of trial and error and massively expand it. Neural networks are an attempt to in some way replicate the operations of neurons in a brain. Various nodes (neurons) receive input, perform some mathematical transformation on that input, and either "fire" (pass some output on to a next step) or not, depending on whether the result of the calculation passes a certain threshold.  Each node starts off with random weights that determine how to mathematically transform the data it receives, and the computer tries many combinations of weights to see which ones are the most accurate. 
+
+
+An Example
+-----
+
+Let's consider a binary question.  Will a given person buy takeout for dinner?  Given seven input variables, you want a single outcome: 1 for "yes, they will get takeout" and 0 for "no, they will not get takeout."
+
+In this image (created thanks to [NN-SVG](http://alexlenail.me/NN-SVG/index.html)), the leftmost column of circles represent **inputs**.  Maybe one input is an individual's level of tiredness, another is the number of days they have had takeout in the last week, a third input could be the number of leftovers in the person's fridge.  These inputs are scaled and normalized.
```

Comment:
```suggestion
In following image (created thanks to [NN-SVG](http://alexlenail.me/NN-SVG/index.html)), the leftmost column of circles represent **inputs**.  Maybe one input is an individual's level of tiredness, another is the number of days they have had takeout in the last week, a third input could be the number of leftovers in the person's fridge.  These inputs are scaled and normalized.
```

just as a cue for non-sighted learners that they haven't missed anything yet, since we're already launching into contextualization here.